File: TP1/Palindromo.java
public class Palindromo {
	
	public static boolean isFim(String palavra){
	
		return(palavra.length() >= 3 && palavra.charAt(0) == 'F' && palavra.charAt(1) == 'I' && palavra.charAt(2)== 'M');
	}


	public static boolean palindromo(String p){
	
		String palindromo = "";

		for(int i = (p.length() - 1); i >= 0; i--){
			palindromo = palindromo + p.charAt(i);
		}

		if(palindromo.equals(p.toLowerCase())){
			return true;
		}else{
			return false;
		}
	}


	public static void main (String[] args){
	
		String[] palavras = new String[10000];
		int qtd = 0;

		do{
		palavras[qtd] = MyIO.readLine();
		
		}while(isFim(palavras[qtd++]) == false);
		qtd--;

		for(int i = 0; i < qtd; i++){
			if(palindromo(palavras[i]) == true){
			MyIO.println("SIM");
			}else{
			MyIO.println("NAO");
			}
		}
	}

}

File: TP1/Palindromo.cpp
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <iostream>
#include <ios>
#include <limits>

using namespace std;

char isFim(string str){

	int tamanho = str.length();

	if(tamanho >= 3 && str[0] == 'F' && str[1] == 'I' && str[2] == 'M'){
		return 1;
	}else{
		return 0;
	}
}

void palindromo(string str){
	int tamanho = str.length();
	string palavra;

	for(int i = (tamanho - 1); i >= 0; i--){
	
		palavra = palavra + str[i];

	}



	if(palavra == str){		
		
		cout << "SIM\n";
	
	}else{

		cout << "NAO\n";
	}
}


int main(){

	int cont = 0;
	string palavra[1000];

	do{
	
	getline(cin, palavra[cont]);

	}while(isFim(palavra[cont++]) != 1);
	cont--;

	for(int i = 0; i < cont; i++){
		palindromo(palavra[i]);
	}	

}
